File: Software/C/binary.c
/*
Compiling and Running in Linux:

 - Open a terminal
 - Navigate to the same directory as binary.c
      cd ~/Programs/jkwalter.github.io/Software/C
 - Type the following to complie binary.c
      gcc binary.c
 - Type the following to run binary.c
      ./a.out
*/

#include <stdio.h>
#include <stdlib.h>

typedef struct linklist *node;

struct linklist
{
 int digit;
 node next;
};

node push(node list, int value)
{
 node tmp;

 tmp = malloc(sizeof(node));
 tmp->digit = value;
 tmp->next = list;

 return tmp;
}

void pop(node list)
{
 node tmp;

 while (list != NULL)
  {
   printf("%d", list->digit);
   tmp = list;
   list = list->next;
   free(tmp);
  }
}

int main(void)
{
 int num;
 node list;

 list = NULL;

 do
  {
   printf("Enter any positive number: ");
   scanf("%d", &num);
  } while (num < 1);

 printf("\n");
 printf("Decimal number: %d\n", num);
 printf("Binary conversion: ");

 while (num != 0)
  {
   list = push(list, num % 2);
   num = num / 2;
  }

 pop(list);

 printf("\n");

 return 0;
}

File: Software/C#/binary.cs
/*
Compiling and Running in Microsoft Visual Studio:

 - Open Developer Command Prompt for Visual Studio
 - Navigate to the same directory as binary.cs
      cd "C:\Users\Josh Walter\Documents\Visual Studio 2017\Projects"
 - Type the following to compile binary.cs
      csc binary.cs
 - Type the following to run binary.cs
      binary
*/

class binary
{
	public static void Main()
	{
        int decimal_number;
        System.Collections.Generic.List<int> binary_number = new System.Collections.Generic.List<int>();

        do
        {
            System.Console.Write("Enter a number greater than 0: ");
            decimal_number = int.Parse(System.Console.ReadLine());
        } while (decimal_number < 1);

        System.Console.WriteLine();

        System.Console.WriteLine("Decimal Number: {0}", decimal_number);

        while (decimal_number != 0)
        {
            binary_number.Add(decimal_number % 2);
            decimal_number = decimal_number / 2;
        }

        System.Console.Write("Binary Representation: ");

        binary_number.Reverse();

        foreach(int digit in binary_number)
        {
            System.Console.Write(digit);
        }

        System.Console.WriteLine();

        binary_number.Clear();
	}
}
